### DISCUSSION TITLE
input-remapper-xautopresets

### DISCUSSION THREAD

**Discussion by DreadPirateLynx** [sezanzeb/input-remapper#294] [Show and tell]
Hey all. I spent the last couple of days writing an automatic profile manager for input-remapper in bash. current version of the script is available [here](https://github.com/DreadPirateLynx/input-remapper-xautopresets/releases).

**Comment by sezanzeb** [comment]
Cool!

Consider adding a .service file, so that it gets started automatically by systemd. They are located in `/usr/lib/systemd/system` and controlled via `systemctl`, if you didn't know this already.

And maybe an installation script that copies everything to where it belongs

**Reply by DreadPirateLynx** [reply]
I'll definitely look into the systemd service. Right now I'm starting it in the background with my .bashrc file, but your way is probably better. I just finished rewriting the script to clean up the logic, parallelize device configuration, include optional device specific .conf files that override the global .conf, and include an internal "_Bypass" profile that can be used to turn injection off for specific applications. Only problem is I can't push the update because I can't get github's authentication working now that they've stopped supporting username/password authentication. I'm sure I'll get it figured out eventually

**Comment by sezanzeb** [comment]
I think "input-remapper-x-autoprofile" would be a more descriptive name by the way

**Reply by DreadPirateLynx** [reply]
For some stupid reason it took me entirely too long to figure out where that “x” was coming from. I see your point

And to anyone using Garuda Linux: I don’t quite understand what the deal with Wayland and X11 is in the flagship version; Wayland’s installed on my system and seems to be running, but I also have access to `xdotool`. If that’s the boat you find yourself in, this will work. EDIT: I'm guessing it's thanks to Kwin

**Comment by DreadPirateLynx** [comment]
Completely re-wrote the script. Added a bunch of features, including running as a user service. Renamed project as suggested.
available [here](https://github.com/DreadPirateLynx/input-remapper-xautopresets)  

Also: is there any way to get `input-remapper-control` to report the injection state/active preset on a given device? I'm having to do some funky things to parallelize device configuration while keeping track of their active presets. This script could be quite a bit leaner if I could just ask `input-remapper-control` what they were instead of tracking them myself.

**Reply by sezanzeb** [reply]
If you'd like, I'll put a link to it into usage.md

**Reply by DreadPirateLynx** [reply]
feel free

**Comment by DreadPirateLynx** [comment]
This is the last time I intend to post here about this, but I've finally gotten to a 1.0 release of this script. I've consolidated the multiple .conf files into a single .ini file, made some significant efficiency improvements to the logic, added the ability to tell the script to ignore certain devices or windows, and added some program settings so the user can have greater control over how the script functions.

[v1.0.0](https://github.com/DreadPirateLynx/input-remapper-xautopresets/releases/tag/v1.0.0)  

If anyone has any comments or suggestions, please feel free to post in either the issues or discussions sections of my project’s github page. I see people have been downloading, but I haven’t been getting any feedback. I’m mostly choosing to take that as “no news is good news,” but it’s always nice to have some confirmation one way or the other. I’m hoping for this to be the go-to solution for anyone wanting automatic preset switching in input-remapper, but it’s hard to be sure I’m on the right path without feedback
